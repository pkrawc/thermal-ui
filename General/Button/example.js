import React from 'react'
import { Button } from '../Button'
import { colors } from 'variables'

export default {
  title: 'Button',
  examples: [
    (
      <Button>Hello World</Button>
    ),
    (
      <Button color={`#DD0000`}>Hello Color</Button>
    ),
    (
      <Button primary color={colors.primary}>Hello Primary</Button>
    )
  ]
}
