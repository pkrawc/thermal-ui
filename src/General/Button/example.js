import React from 'react'
import { Button } from '../Button'
import { colors } from 'variables'

const ButtonExample = props =>
  <Button primary color={colors.primary} onClick={() => { console.log('button click')}}>
    Hello Button
  </Button>

export default {
  title: 'Button',
  example: ButtonExample
}
