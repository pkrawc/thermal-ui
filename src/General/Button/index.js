import styled from 'styled-components'
import { colors, shadows } from 'variables'

export const Button = styled.button`
  padding: 0.5rem 1rem;
  background-color: ${({primary, color}) => {
      if (primary) {
        if (color) return color
        else return colors.dark
      } else {
        return colors.light
      }
    }
  };
  border: 2px solid ${({color}) => color ? color : colors.dark};
  border-radius: 2px;
  box-shadow: none;
  color: ${
    ({color, primary}) => {
      if (primary) return colors.light
      else {
        if (color) return color
        else return colors.dark
      }
    }
  };
  cursor: pointer;
  font-size: 1rem;
  font-weight: 600;
  min-width: 4rem;
  outline: none;
  text-align: center;
  text-transform: uppercase;
  transition: all 150ms ease-in-out;
  white-space: nowrap;
  &:hover {
    background-color: ${
      ({primary, color}) => {
        if (primary) {
          return colors.light
        } else {
          if (color) return color
          else return colors.dark
        }
      }
    };
    box-shadow: ${shadows.medium};
    color: ${
      ({primary, color}) => {
        if (primary) {
          if (color) return color
          else return colors.dark
        }
        else return colors.light
      }
    };
  }
  &:focus: {
    box-shadow: none;
  }
`
