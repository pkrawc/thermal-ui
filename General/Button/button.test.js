import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

describe('Button', () => {

  it('renders a Button without crashing', () => {
    const tree = renderer.create(
      <Button>Hello World</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })

})
