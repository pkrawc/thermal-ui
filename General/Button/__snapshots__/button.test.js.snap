exports[`Button renders a Button without crashing 1`] = `
<button
  className="EuKFK"
  onClick={undefined}>
  <span>
    Hello World
  </span>
</button>
`;
